// ionic-angular/components/tabs/tab.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/tabs/tab';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/components/tabs/tabs';
import * as import9 from 'ionic-angular/components/app/app';
import * as import10 from 'ionic-angular/config/config';
import * as import11 from 'ionic-angular/util/keyboard';
import * as import12 from '@angular/core/src/linker/element_ref';
import * as import13 from '@angular/core/src/zone/ng_zone';
import * as import14 from '@angular/core/src/linker/component_factory_resolver';
import * as import15 from 'ionic-angular/gestures/gesture-controller';
import * as import16 from 'ionic-angular/transitions/transition-controller';
import * as import17 from 'ionic-angular/navigation/deep-linker';
import * as import18 from '@angular/core/src/metadata/view';
import * as import19 from '@angular/core/src/linker/component_factory';
import * as import20 from '@angular/core/src/linker/query_list';
var renderType_Tab_Host = null;
var _View_Tab_Host0 = (function (_super) {
    __extends(_View_Tab_Host0, _super);
    function _View_Tab_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Tab_Host0, renderType_Tab_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Tab_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-tab', rootSelector, null);
        this.renderer.setElementAttribute(this._el_0, 'role', 'tabpanel');
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Tab0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Tab_0_4 = new import3.Tab(this.parentInjector.get(import8.Tabs), this.parentInjector.get(import9.App), this.parentInjector.get(import10.Config), this.parentInjector.get(import11.Keyboard), new import12.ElementRef(this._el_0), this.parentInjector.get(import13.NgZone), this.renderer, this.parentInjector.get(import14.ComponentFactoryResolver), compView_0.ref, this.parentInjector.get(import15.GestureController), this.parentInjector.get(import16.TransitionController), this.parentInjector.get(import17.DeepLinker, null));
        this._appEl_0.initComponent(this._Tab_0_4, [], compView_0);
        compView_0.create(this._Tab_0_4, this.projectableNodes, null);
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Tab_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.Tab) && (0 === requestNodeIndex))) {
            return this._Tab_0_4;
        }
        return notFoundResult;
    };
    _View_Tab_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (((this.numberOfChecks === 0) && !throwOnChange)) {
            this._Tab_0_4.ngOnInit();
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = this._Tab_0_4._tabId;
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementAttribute(this._el_0, 'id', ((currVal_0 == null) ? null : currVal_0.toString()));
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this._Tab_0_4._btnId;
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-labelledby', ((currVal_1 == null) ? null : currVal_1.toString()));
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Tab_Host0;
}(import1.AppView));
function viewFactory_Tab_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Tab_Host === null)) {
        (renderType_Tab_Host = viewUtils.createRenderComponentType('', 0, import18.ViewEncapsulation.None, [], {}));
    }
    return new _View_Tab_Host0(viewUtils, parentInjector, declarationEl);
}
export var TabNgFactory = new import19.ComponentFactory('ion-tab', viewFactory_Tab_Host0, import3.Tab);
var styles_Tab = [];
var renderType_Tab = null;
var _View_Tab0 = (function (_super) {
    __extends(_View_Tab0, _super);
    function _View_Tab0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Tab0, renderType_Tab, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Tab0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._viewQuery_viewport_0 = new import20.QueryList();
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        this._el_1 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'nav-decor');
        this._viewQuery_viewport_0.reset([this._appEl_0.vcRef]);
        this.context._vp = this._viewQuery_viewport_0.first;
        this.init([], [
            this._el_0,
            this._el_1
        ], [], []);
        return null;
    };
    return _View_Tab0;
}(import1.AppView));
export function viewFactory_Tab0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Tab === null)) {
        (renderType_Tab = viewUtils.createRenderComponentType('/Users/Perkka/booster/node_modules/ionic-angular/components/tabs/tab.d.ts class Tab - inline template', 0, import18.ViewEncapsulation.None, styles_Tab, {}));
    }
    return new _View_Tab0(viewUtils, parentInjector, declarationEl);
}
