// ionic-angular/components/tabs/tabs.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/tabs/tabs';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/navigation/nav-controller';
import * as import9 from 'ionic-angular/navigation/view-controller';
import * as import10 from 'ionic-angular/components/app/app';
import * as import11 from 'ionic-angular/config/config';
import * as import12 from '@angular/core/src/linker/element_ref';
import * as import13 from 'ionic-angular/platform/platform';
import * as import14 from 'ionic-angular/navigation/deep-linker';
import * as import15 from '@angular/core/src/metadata/view';
import * as import16 from '@angular/core/src/linker/component_factory';
import * as import17 from '@angular/core/src/linker/query_list';
import * as import18 from '@angular/common/src/directives/ng_for';
import * as import19 from '@angular/core/src/linker/template_ref';
import * as import20 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import21 from 'ionic-angular/components/tabs/tab-button';
import * as import22 from '@angular/common/src/directives/ng_if';
import * as import23 from 'ionic-angular/components/icon/icon';
import * as import24 from '@angular/common/src/directives/ng_class';
import * as import25 from 'ionic-angular/components/badge/badge';
import * as import26 from '@angular/core/src/change_detection/differs/keyvalue_differs';
var renderType_Tabs_Host = null;
var _View_Tabs_Host0 = (function (_super) {
    __extends(_View_Tabs_Host0, _super);
    function _View_Tabs_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Tabs_Host0, renderType_Tabs_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Tabs_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-tabs', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Tabs0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Tabs_0_4 = new import3.Tabs(this.parentInjector.get(import8.NavController, null), this.parentInjector.get(import9.ViewController, null), this.parentInjector.get(import10.App), this.parentInjector.get(import11.Config), new import12.ElementRef(this._el_0), this.parentInjector.get(import13.Platform), this.renderer, this.parentInjector.get(import14.DeepLinker));
        this._appEl_0.initComponent(this._Tabs_0_4, [], compView_0);
        compView_0.create(this._Tabs_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Tabs_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.Tabs) && (0 === requestNodeIndex))) {
            return this._Tabs_0_4;
        }
        return notFoundResult;
    };
    _View_Tabs_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if ((this.numberOfChecks === 0)) {
                this._Tabs_0_4.ngAfterViewInit();
            }
        }
    };
    return _View_Tabs_Host0;
}(import1.AppView));
function viewFactory_Tabs_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Tabs_Host === null)) {
        (renderType_Tabs_Host = viewUtils.createRenderComponentType('', 0, import15.ViewEncapsulation.None, [], {}));
    }
    return new _View_Tabs_Host0(viewUtils, parentInjector, declarationEl);
}
export var TabsNgFactory = new import16.ComponentFactory('ion-tabs', viewFactory_Tabs_Host0, import3.Tabs);
var styles_Tabs = [];
var renderType_Tabs = null;
var _View_Tabs0 = (function (_super) {
    __extends(_View_Tabs0, _super);
    function _View_Tabs0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Tabs0, renderType_Tabs, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Tabs0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._viewQuery_TabHighlight_0 = new import17.QueryList();
        this._viewQuery_tabbar_1 = new import17.QueryList();
        this._viewQuery_portal_2 = new import17.QueryList();
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'tabbar');
        this.renderer.setElementAttribute(this._el_0, 'role', 'tablist');
        this._anchor_1 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_1 = new import2.AppElement(1, 0, this, this._anchor_1);
        this._TemplateRef_1_5 = new import19.TemplateRef_(this._appEl_1, viewFactory_Tabs1);
        this._NgFor_1_6 = new import18.NgFor(this._appEl_1.vcRef, this._TemplateRef_1_5, this.parentInjector.get(import20.IterableDiffers), this.ref);
        this._el_2 = this.renderer.createElement(this._el_0, 'div', null);
        this.renderer.setElementAttribute(this._el_2, 'class', 'tab-highlight');
        this.renderer.projectNodes(parentRenderNode, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._el_3 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_3, 'tab-portal', '');
        this._appEl_3 = new import2.AppElement(3, null, this, this._el_3);
        this._expr_0 = import7.UNINITIALIZED;
        this._viewQuery_TabHighlight_0.reset([]);
        this.context._highlight = this._viewQuery_TabHighlight_0.first;
        this._viewQuery_tabbar_1.reset([new import12.ElementRef(this._el_0)]);
        this.context._tabbar = this._viewQuery_tabbar_1.first;
        this._viewQuery_portal_2.reset([this._appEl_3.vcRef]);
        this.context.portal = this._viewQuery_portal_2.first;
        this.init([], [
            this._el_0,
            this._anchor_1,
            this._el_2,
            this._el_3
        ], [], []);
        return null;
    };
    _View_Tabs0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import19.TemplateRef) && (1 === requestNodeIndex))) {
            return this._TemplateRef_1_5;
        }
        if (((token === import18.NgFor) && (1 === requestNodeIndex))) {
            return this._NgFor_1_6;
        }
        return notFoundResult;
    };
    _View_Tabs0.prototype.detectChangesInternal = function (throwOnChange) {
        var changes = null;
        changes = null;
        var currVal_0 = this.context._tabs;
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this._NgFor_1_6.ngForOf = currVal_0;
            if ((changes === null)) {
                (changes = {});
            }
            changes['ngForOf'] = new import7.SimpleChange(this._expr_0, currVal_0);
            this._expr_0 = currVal_0;
        }
        if ((changes !== null)) {
            this._NgFor_1_6.ngOnChanges(changes);
        }
        if (!throwOnChange) {
            this._NgFor_1_6.ngDoCheck();
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Tabs0;
}(import1.AppView));
export function viewFactory_Tabs0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Tabs === null)) {
        (renderType_Tabs = viewUtils.createRenderComponentType('/Users/Perkka/booster/node_modules/ionic-angular/components/tabs/tabs.d.ts class Tabs - inline template', 1, import15.ViewEncapsulation.None, styles_Tabs, {}));
    }
    return new _View_Tabs0(viewUtils, parentInjector, declarationEl);
}
var _View_Tabs1 = (function (_super) {
    __extends(_View_Tabs1, _super);
    function _View_Tabs1(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Tabs1, renderType_Tabs, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Tabs1.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'a', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'tab-button');
        this.renderer.setElementAttribute(this._el_0, 'href', '#');
        this.renderer.setElementAttribute(this._el_0, 'role', 'tab');
        this._TabButton_0_3 = new import21.TabButton(this.parent.parentInjector.get(import11.Config), new import12.ElementRef(this._el_0), this.renderer);
        this._anchor_1 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_1 = new import2.AppElement(1, 0, this, this._anchor_1);
        this._TemplateRef_1_5 = new import19.TemplateRef_(this._appEl_1, viewFactory_Tabs2);
        this._NgIf_1_6 = new import22.NgIf(this._appEl_1.vcRef, this._TemplateRef_1_5);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_2 = new import2.AppElement(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import19.TemplateRef_(this._appEl_2, viewFactory_Tabs3);
        this._NgIf_2_6 = new import22.NgIf(this._appEl_2.vcRef, this._TemplateRef_2_5);
        this._anchor_3 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_3 = new import2.AppElement(3, 0, this, this._anchor_3);
        this._TemplateRef_3_5 = new import19.TemplateRef_(this._appEl_3, viewFactory_Tabs4);
        this._NgIf_3_6 = new import22.NgIf(this._appEl_3.vcRef, this._TemplateRef_3_5);
        this._el_4 = this.renderer.createElement(this._el_0, 'div', null);
        this.renderer.setElementAttribute(this._el_4, 'class', 'button-effect');
        this._expr_2 = import7.UNINITIALIZED;
        this._expr_3 = import7.UNINITIALIZED;
        var disposable_0 = this.renderer.listen(this._el_0, 'ionSelect', this.eventHandler(this._handle_ionSelect_0_0.bind(this)));
        var disposable_1 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_1.bind(this)));
        this._expr_4 = import7.UNINITIALIZED;
        this._expr_5 = import7.UNINITIALIZED;
        this._expr_6 = import7.UNINITIALIZED;
        this._expr_7 = import7.UNINITIALIZED;
        this._expr_8 = import7.UNINITIALIZED;
        this._expr_9 = import7.UNINITIALIZED;
        this._expr_10 = import7.UNINITIALIZED;
        this._expr_11 = import7.UNINITIALIZED;
        this._expr_12 = import7.UNINITIALIZED;
        this._expr_13 = import7.UNINITIALIZED;
        var subscription_0 = this._TabButton_0_3.ionSelect.subscribe(this.eventHandler(this._handle_ionSelect_0_0.bind(this)));
        this._expr_14 = import7.UNINITIALIZED;
        this._expr_15 = import7.UNINITIALIZED;
        this._expr_16 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._anchor_1,
            this._anchor_2,
            this._anchor_3,
            this._el_4
        ], [
            disposable_0,
            disposable_1
        ], [subscription_0]);
        return null;
    };
    _View_Tabs1.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import19.TemplateRef) && (1 === requestNodeIndex))) {
            return this._TemplateRef_1_5;
        }
        if (((token === import22.NgIf) && (1 === requestNodeIndex))) {
            return this._NgIf_1_6;
        }
        if (((token === import19.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import22.NgIf) && (2 === requestNodeIndex))) {
            return this._NgIf_2_6;
        }
        if (((token === import19.TemplateRef) && (3 === requestNodeIndex))) {
            return this._TemplateRef_3_5;
        }
        if (((token === import22.NgIf) && (3 === requestNodeIndex))) {
            return this._NgIf_3_6;
        }
        if (((token === import21.TabButton) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 4)))) {
            return this._TabButton_0_3;
        }
        return notFoundResult;
    };
    _View_Tabs1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_4 = this.context.$implicit;
        if (import4.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this._TabButton_0_3.tab = currVal_4;
            this._expr_4 = currVal_4;
        }
        if (((this.numberOfChecks === 0) && !throwOnChange)) {
            this._TabButton_0_3.ngOnInit();
        }
        var currVal_14 = this.context.$implicit.tabIcon;
        if (import4.checkBinding(throwOnChange, this._expr_14, currVal_14)) {
            this._NgIf_1_6.ngIf = currVal_14;
            this._expr_14 = currVal_14;
        }
        var currVal_15 = this.context.$implicit.tabTitle;
        if (import4.checkBinding(throwOnChange, this._expr_15, currVal_15)) {
            this._NgIf_2_6.ngIf = currVal_15;
            this._expr_15 = currVal_15;
        }
        var currVal_16 = this.context.$implicit.tabBadge;
        if (import4.checkBinding(throwOnChange, this._expr_16, currVal_16)) {
            this._NgIf_3_6.ngIf = currVal_16;
            this._expr_16 = currVal_16;
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_2 = !this.context.$implicit.enabled;
        if (import4.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementClass(this._el_0, 'tab-disabled', currVal_2);
            this._expr_2 = currVal_2;
        }
        var currVal_3 = !this.context.$implicit.show;
        if (import4.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this.renderer.setElementClass(this._el_0, 'tab-hidden', currVal_3);
            this._expr_3 = currVal_3;
        }
        var currVal_5 = this._TabButton_0_3.tab._btnId;
        if (import4.checkBinding(throwOnChange, this._expr_5, currVal_5)) {
            this.renderer.setElementAttribute(this._el_0, 'id', ((currVal_5 == null) ? null : currVal_5.toString()));
            this._expr_5 = currVal_5;
        }
        var currVal_6 = this._TabButton_0_3.tab._tabId;
        if (import4.checkBinding(throwOnChange, this._expr_6, currVal_6)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-controls', ((currVal_6 == null) ? null : currVal_6.toString()));
            this._expr_6 = currVal_6;
        }
        var currVal_7 = this._TabButton_0_3.tab.isSelected;
        if (import4.checkBinding(throwOnChange, this._expr_7, currVal_7)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-selected', ((currVal_7 == null) ? null : currVal_7.toString()));
            this._expr_7 = currVal_7;
        }
        var currVal_8 = this._TabButton_0_3.hasTitle;
        if (import4.checkBinding(throwOnChange, this._expr_8, currVal_8)) {
            this.renderer.setElementClass(this._el_0, 'has-title', currVal_8);
            this._expr_8 = currVal_8;
        }
        var currVal_9 = this._TabButton_0_3.hasIcon;
        if (import4.checkBinding(throwOnChange, this._expr_9, currVal_9)) {
            this.renderer.setElementClass(this._el_0, 'has-icon', currVal_9);
            this._expr_9 = currVal_9;
        }
        var currVal_10 = this._TabButton_0_3.hasTitleOnly;
        if (import4.checkBinding(throwOnChange, this._expr_10, currVal_10)) {
            this.renderer.setElementClass(this._el_0, 'has-title-only', currVal_10);
            this._expr_10 = currVal_10;
        }
        var currVal_11 = this._TabButton_0_3.hasIconOnly;
        if (import4.checkBinding(throwOnChange, this._expr_11, currVal_11)) {
            this.renderer.setElementClass(this._el_0, 'icon-only', currVal_11);
            this._expr_11 = currVal_11;
        }
        var currVal_12 = this._TabButton_0_3.hasBadge;
        if (import4.checkBinding(throwOnChange, this._expr_12, currVal_12)) {
            this.renderer.setElementClass(this._el_0, 'has-badge', currVal_12);
            this._expr_12 = currVal_12;
        }
        var currVal_13 = this._TabButton_0_3.disHover;
        if (import4.checkBinding(throwOnChange, this._expr_13, currVal_13)) {
            this.renderer.setElementClass(this._el_0, 'disable-hover', currVal_13);
            this._expr_13 = currVal_13;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_Tabs1.prototype._handle_ionSelect_0_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.parent.context.select($event) !== false);
        return (true && pd_0);
    };
    _View_Tabs1.prototype._handle_click_0_1 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this._TabButton_0_3.onClick() !== false);
        return (true && pd_0);
    };
    return _View_Tabs1;
}(import1.AppView));
function viewFactory_Tabs1(viewUtils, parentInjector, declarationEl) {
    return new _View_Tabs1(viewUtils, parentInjector, declarationEl);
}
var _View_Tabs2 = (function (_super) {
    __extends(_View_Tabs2, _super);
    function _View_Tabs2(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Tabs2, renderType_Tabs, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Tabs2.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'ion-icon', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'tab-button-icon');
        this.renderer.setElementAttribute(this._el_0, 'role', 'img');
        this._Icon_0_3 = new import23.Icon(this.parent.parent.parentInjector.get(import11.Config), new import12.ElementRef(this._el_0), this.renderer);
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this._expr_2 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return null;
    };
    _View_Tabs2.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import23.Icon) && (0 === requestNodeIndex))) {
            return this._Icon_0_3;
        }
        return notFoundResult;
    };
    _View_Tabs2.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = this.parent.context.$implicit.tabIcon;
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this._Icon_0_3.name = currVal_0;
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this.parent.context.$implicit.isSelected;
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this._Icon_0_3.isActive = currVal_1;
            this._expr_1 = currVal_1;
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_2 = this._Icon_0_3._hidden;
        if (import4.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementClass(this._el_0, 'hide', currVal_2);
            this._expr_2 = currVal_2;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_Tabs2.prototype.destroyInternal = function () {
        this._Icon_0_3.ngOnDestroy();
    };
    return _View_Tabs2;
}(import1.AppView));
function viewFactory_Tabs2(viewUtils, parentInjector, declarationEl) {
    return new _View_Tabs2(viewUtils, parentInjector, declarationEl);
}
var _View_Tabs3 = (function (_super) {
    __extends(_View_Tabs3, _super);
    function _View_Tabs3(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Tabs3, renderType_Tabs, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Tabs3.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'span', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'tab-button-text');
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this._expr_0 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1
        ], [], []);
        return null;
    };
    _View_Tabs3.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = import4.interpolate(1, '', this.parent.context.$implicit.tabTitle, '');
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setText(this._text_1, currVal_0);
            this._expr_0 = currVal_0;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Tabs3;
}(import1.AppView));
function viewFactory_Tabs3(viewUtils, parentInjector, declarationEl) {
    return new _View_Tabs3(viewUtils, parentInjector, declarationEl);
}
var _View_Tabs4 = (function (_super) {
    __extends(_View_Tabs4, _super);
    function _View_Tabs4(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Tabs4, renderType_Tabs, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Tabs4.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'ion-badge', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'tab-badge');
        this._NgClass_0_3 = new import24.NgClass(this.parent.parent.parentInjector.get(import20.IterableDiffers), this.parent.parent.parentInjector.get(import26.KeyValueDiffers), new import12.ElementRef(this._el_0), this.renderer);
        this._Badge_0_4 = new import25.Badge(this.parent.parent.parentInjector.get(import11.Config), new import12.ElementRef(this._el_0), this.renderer);
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this._expr_2 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1
        ], [], []);
        return null;
    };
    _View_Tabs4.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import24.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 1)))) {
            return this._NgClass_0_3;
        }
        if (((token === import25.Badge) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 1)))) {
            return this._Badge_0_4;
        }
        return notFoundResult;
    };
    _View_Tabs4.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = 'tab-badge';
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this._NgClass_0_3.klass = currVal_0;
            this._expr_0 = currVal_0;
        }
        var currVal_1 = ('badge-' + this.parent.context.$implicit.tabBadgeStyle);
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this._NgClass_0_3.ngClass = currVal_1;
            this._expr_1 = currVal_1;
        }
        if (!throwOnChange) {
            this._NgClass_0_3.ngDoCheck();
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_2 = import4.interpolate(1, '', this.parent.context.$implicit.tabBadge, '');
        if (import4.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setText(this._text_1, currVal_2);
            this._expr_2 = currVal_2;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Tabs4;
}(import1.AppView));
function viewFactory_Tabs4(viewUtils, parentInjector, declarationEl) {
    return new _View_Tabs4(viewUtils, parentInjector, declarationEl);
}
