// ionic-angular/components/scroll/scroll.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/scroll/scroll';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/element_ref';
import * as import9 from '@angular/core/src/metadata/view';
import * as import10 from '@angular/core/src/linker/component_factory';
var renderType_Scroll_Host = null;
var _View_Scroll_Host0 = (function (_super) {
    __extends(_View_Scroll_Host0, _super);
    function _View_Scroll_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Scroll_Host0, renderType_Scroll_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Scroll_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-scroll', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Scroll0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Scroll_0_4 = new import3.Scroll(new import8.ElementRef(this._el_0));
        this._appEl_0.initComponent(this._Scroll_0_4, [], compView_0);
        compView_0.create(this._Scroll_0_4, this.projectableNodes, null);
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Scroll_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.Scroll) && (0 === requestNodeIndex))) {
            return this._Scroll_0_4;
        }
        return notFoundResult;
    };
    _View_Scroll_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (((this.numberOfChecks === 0) && !throwOnChange)) {
            this._Scroll_0_4.ngOnInit();
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = this._Scroll_0_4.scrollX;
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementClass(this._el_0, 'scroll-x', currVal_0);
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this._Scroll_0_4.scrollY;
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementClass(this._el_0, 'scroll-y', currVal_1);
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Scroll_Host0;
}(import1.AppView));
function viewFactory_Scroll_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Scroll_Host === null)) {
        (renderType_Scroll_Host = viewUtils.createRenderComponentType('', 0, import9.ViewEncapsulation.None, [], {}));
    }
    return new _View_Scroll_Host0(viewUtils, parentInjector, declarationEl);
}
export var ScrollNgFactory = new import10.ComponentFactory('ion-scroll', viewFactory_Scroll_Host0, import3.Scroll);
var styles_Scroll = [];
var renderType_Scroll = null;
var _View_Scroll0 = (function (_super) {
    __extends(_View_Scroll0, _super);
    function _View_Scroll0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Scroll0, renderType_Scroll, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckOnce);
    }
    _View_Scroll0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'scroll-content');
        this._el_1 = this.renderer.createElement(this._el_0, 'div', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'scroll-zoom-wrapper');
        this.renderer.projectNodes(this._el_1, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this.init([], [
            this._el_0,
            this._el_1
        ], [], []);
        return null;
    };
    return _View_Scroll0;
}(import1.AppView));
export function viewFactory_Scroll0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Scroll === null)) {
        (renderType_Scroll = viewUtils.createRenderComponentType('/Users/Perkka/booster/node_modules/ionic-angular/components/scroll/scroll.d.ts class Scroll - inline template', 1, import9.ViewEncapsulation.None, styles_Scroll, {}));
    }
    return new _View_Scroll0(viewUtils, parentInjector, declarationEl);
}
